zio-http/src/main/scala/zio/http/ClientSSLConfig.scala

```diff
@@ -21,25 +21,32 @@ import zio.Config
 sealed trait ClientSSLConfig
 
 object ClientSSLConfig {
+  val config: Config[ClientSSLConfig] = {
+    val tpe                = Config.string("type")
+    val certPath           = Config.string("certPath")
+    val trustStorePath     = Config.string("trustStorePath")
+    val trustStorePassword = Config.secret("trustStorePassword").map(d => new String(d.value.toArray))
+
+    val default                = Config.succeed(Default)
+    val fromCertFile           = certPath.map(FromCertFile(_))
+    val fromCertResource       = certPath.map(FromCertResource(_))
+    val fromTrustStoreFile     = trustStorePath.zipWith(trustStorePassword)(FromTrustStoreFile(_, _))
+    val fromTrustStoreResource = trustStorePath.zipWith(trustStorePassword)(FromTrustStoreResource(_, _))
+
+    tpe
+      .switch(
+        "Default"                -> default,
+        "FromCertFile"           -> fromCertFile,
+        "FromCertResource"       -> fromCertResource,
+        "FromTrustStoreFile"     -> fromTrustStoreFile,
+        "FromTrustStoreResource" -> fromTrustStoreResource,
+      )
+      .nested("zio.http.ClientSSLConfig")
```

Comment:
I think this nesting is not good the way this is used now in `ZClient.Config` (where it gets explicitly `nested` too)

Reply:
👍